libbindgen/build.rs

```diff
@@ -14,6 +13,42 @@ mod codegen {
     }
 }
 
+mod testgen {
+    use std::char;
+    use std::ffi::OsStr;
+    use std::fs::{self, File};
+    use std::io::Write;
+    use std::path::Path;
+
+    pub fn main() {
+        let out_dir = Path::new(env!("OUT_DIR"));
+        let mut dst = File::create(Path::new(&out_dir).join("tests.rs")).unwrap();
+
+        let manifest_dir = Path::new(env!("CARGO_MANIFEST_DIR"));
+        let headers_dir = manifest_dir.join("tests").join("headers");
+
+        let entries = fs::read_dir(headers_dir)
+            .expect("Couldn't read headers dir")
+            .map(|result| result.expect("Couldn't read header file"));
+
+        for entry in entries {
+            match entry.path().extension().and_then(OsStr::to_str) {
+                Some("h") | Some("hpp") => {
+                    let func = entry.file_name().to_str().unwrap()
+                        .replace(|c| !char::is_alphanumeric(c), "_")
+                        .replace("__", "_")
+                        .to_lowercase();
+                    let _ = writeln!(dst, "test_header!(header_{}, {:?});",
```

Comment:
I *think* You need to write these in a block to avoid https://github.com/rust-lang/rust/issues/35560